pyarlo/base_station.py

```diff
@@ -113,7 +301,11 @@ def mode(self, mode):
         """
         if mode not in ACTION_MODES.keys():
             return "Invalid mode"
-        return self.__run_action(mode)
+        return self.__run_action(
+            method='SET',
+            resource='modes',
+            mode=mode,
+            publish_response=True)
```

Comment:
make sure to call `update()` method after changing the mode otherwise it will not update. For example:

```python
In [4]: base.mode
Out[4]: u'armed'

In [5]: base.available_modes
Out[5]: ['schedule', 'armed', 'disarmed', 'custom']

In [6]: base.mode = 'disarmed'

In [7]: base.mode

In [8]: base.update()

In [9]: base.mode
Out[9]: u'disarmed'
```